### PR TITLE
BUG: stats: fix multivariate_hypergeom.rvs method

### LINKED ISSUE #16171: BUG: scipy.stats.multivariate_hypergeom.rvs raises ValueError when at least the last two populations are 0
### Describe your issue.

I ran into a strange issue when sampling using Scipy's multivariate hypergeometric distribution. Calling `scipy.stats.multivariate_hypergeom.rvs` on a list of populations raises the error message `ValueError: ngood + nbad < nsample`, while calling the same function on the sorted array as input does not.
Specfically, the error seems to be raised when at least the last two values of the population are 0. 

### Reproducing Code Example

```python
import scipy.stats
r = scipy.stats.multivariate_hypergeom.rvs
res1 = r([0, 20], 5) # Returns [0, 5]
res2 = r([0, 0, 20], 5) # Returns [0, 0, 5]
res3 = r([0, 0, 20, 0], 5) # # Returns [0, 0, 5, 0]
res4 = r([0, 0, 20, 0, 0], 5) # # Raises error
```

### Error message

```shell
Traceback (most recent call last):
  File "<string>", line 6, in <module>
  File "/home/alexander/miniconda3/envs/py37/lib/python3.7/site-packages/scipy/stats/_multivariate.py", line 4726, in rvs
    size=size))
  File "mtrand.pyx", line 3839, in numpy.random.mtrand.RandomState.hypergeometric
ValueError: ngood + nbad < nsample
```

### SciPy/NumPy/Python version information

1.7.3 1.21.5 sys.version_info(major=3, minor=10, micro=4, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
#### Reference issue

fixes #16171 

#### What does this implement/fix?

Fixes multivariate_hypergeom.rvs method when the last two or more elements of the population are 0.

### CODE REVIEW

File: scipy/stats/tests/test_multivariate.py [hunk 1]
```diff
@@ -1976,6 +1976,17 @@ def test_rvs_broadcasting(self):
         rvs = rv.rvs(size=(1000, 2), random_state=123)
         assert_allclose(rvs.mean(0), rv.mean(), rtol=1e-2)
 
+    @pytest.mark.parametrize('m, n', (
+        ([0, 0, 20, 0, 0], 5), ([0, 0, 0, 0, 0], 0), ([0, 0], 0),
```
Comment (tupui) [inline scipy/stats/tests/test_multivariate.py:1980]: You already have this case in your test matrix. Or you had another case in mind maybe?
```suggestion
        ([0, 0, 20, 0, 0], 5), ([0, 0, 0, 0, 0], 0),
```

**Review by tupui — APPROVED** [review]
Clean change. Thanks @tirthasheshpatel. I will merge after you reply to the suggestion.

File: scipy/stats/tests/test_multivariate.py (hunk 1 above)
Comment (tirthasheshpatel) [reply scipy/stats/tests/test_multivariate.py:1980]: > Or you had another case in mind maybe?

I just wanted to make sure that small arrays with all 0s don't error out.

Comment (tirthasheshpatel) [reply scipy/stats/tests/test_multivariate.py:1980]: Oh, I see that this test case is just a duplicate of the one below. Will remove it as you have suggested here. Thanks for noticing!

**Pull request merged**